doc/pic/hld_fpmsyncd.md

```diff
@@ -475,6 +515,46 @@ Sample of CONFIG DB snippet given below:
         }
     },
 ```
+#### APPL DB Enhancements  
+
+This feature extends the existing NEXT_HOP_GROUP_TABLE, to store next hop group information to be used by one or more routes. Recursive/ecmp routes are represented by referencing a list of nexthop group ids in the nexthop_group field. When this field is used, other fields will not be present.
+
+
+```
+### NEXT_HOP_GROUP_TABLE
+  Stores a list of groups of one or more next hops
+  Status: Mandatory
+    key           = NEXT_HOP_GROUP_TABLE:string ; 
+                    arbitrary string identifying the next hop 
+                    group, as determined by the programming 
+                    application.
+    nexthop       = *prefix,           ; IP addresses separated
+                     “,” (empty indicates no gateway)
+    ifname        = *PORT_TABLE.key,   ; zero or more separated 
+                     by “,” (zero indicates no interface)
+    nexthop_group = NEXT_HOP_GROUP_TABLE:key, ; index within the
```

Comment:
can you add an example for the recursive routes?

Reply:
The DB entry for recursive NHG is same as ECMP NHGs. In that for recursive NHG may contain only one NH and hence the NHA_GROUP will contain only one NHID